js/index.js

```diff
@@ -37,6 +37,201 @@ const siteContent = {
+});
+
+
+// Append a child to the nav
+const newNode = document.createElement('a'); 
+newNode.appendChild(document.createTextNode('Github'));
+newNode.setAttribute('href', 'https://github.com/greatidea');
+const navAddNode = document.querySelector('nav').appendChild(newNode); 
+
+
+// Prepend a child to the nav
+const newNode2 = document.createElement('a');
+newNode2.setAttribute('href', 'index.html');
+newNode2.appendChild(document.createTextNode('Home'));
+const navAddNode2 = document.querySelector('nav').prepend(newNode2); 
+
+
+const navColorArr = Array.from(document.getElementsByTagName('a'));
+navColorArr.forEach(element => {
+  element.style.color = '#295205';
+});
+
+
+// // Background color just to see if I know how that works
+const html = document.getElementsByTagName('html');
+html[0].style.height = '150%';
+const body = document.getElementsByTagName('body');
+body[0].style = `background-image: linear-gradient(167deg, rgba(207, 217, 226, 0.16) 35%, rgba(42, 76, 83, 0.94));`;
+
+
+// // Things were going fine, but then I had a great idea.. no pun intended!
+const logo = document.querySelector('header #logo-img');
+logo.setAttribute('src', siteContent['nav']['img-src']);
+
+
+// // This is section class='cta'
+// // Traversing over an object literal and html structure. 
+const cta = document.querySelector('.cta'); // This gives us the entire section
+const ctaAsArr = Array.from(cta.children);  // Make an array from the children nodeList method
+ctaAsArr.forEach(element => { // This loop iterates through ctaAsArr to render the correct data from siteContent. Starts at section class cta
+  if (element.localName === 'div') { // next is level is div class cta-text and the img.
+    for (let i=0; i<element.children.length; i++) { // this loop determines which of the two grandchildren nodes it is, and acts accordingly.
+      if (element.children[i].localName === 'h1') { // keep digging!
+        element.children[i].textContent = siteContent['cta']['h1'];
+      } else if (element.children[i].localName === 'button') { 
+        element.children[i].setAttribute('value', siteContent['cta']['button']);
+        element.children[i].textContent = siteContent['cta']['button'];
+      } else {
+        console.log('Oh no! Something went wrong! Grand child');
+      }
+    }
+  } else if (element.localName === 'img') { // this level is the sibling to div
+    element.src = siteContent['cta']['img-src'];
+  } else {
+    console.log('Oh no! Something went wrong! Child');
+  }
+});
+
+
+// // This is section class='main-content'
+// // There is a way, I believe, to make this much more elegant. Instead, I'm left with
+// // if logic that probably would have been more efficient outside of the forEach loop.
+// // Anyways, I've spent too much time on it so I just finished it out 'more simply'.
+const mainContent = document.querySelector('.main-content');
+const mainAsArr = Array.from(mainContent.children);
+mainAsArr.forEach(element => {
```

Comment:
This is certainly fun, but sometimes easier to read > Javascript complexity.

Ideally what we want to do when we use Javascript functions is to add abstraction which makes our functionality more dynamic. Here we're adding abstraction and things are mostly still just hard-coded, so we wrote a TON more syntax and lines of code to achieve basically the same result.